### ISSUE TITLE
Localization

### ISSUE THREAD

**Issue by thefoxcc** [FedericoHeichou/PizzaReader#63]
Hey, buddy.
I have a suggestion you can make js/json localization files Frontend and Backend.

Also suggestion to implement default scrolling type selection when reading manga.

**Comment by FedericoHeichou** [comment]
> Hey, buddy. I have a suggestion you can make js/json localization files Frontend and Backend.
> 
> Also suggestion to implement default scrolling type selection when reading manga.

Hello.  
Frontend has already localizations, the files are 2 and are pretty small, you just need to copy one (for example it.json and it.php) and modify it. It is documented in the readme.  

If you want to change the default scrolling type you just have to change the comic's format from "manga" to "long strip" in the comic's settings.

**Comment by thefoxcc** [comment]
> > Hey, buddy. I have a suggestion you can make js/json localization files Frontend and Backend.
> > Also suggestion to implement default scrolling type selection when reading manga.
> 
> Hello. Frontend has already localizations, the files are 2 and are pretty small, you just need to copy one (for example it.json and it.php) and modify it. It is documented in the readme.
> 
> If you want to change the default scrolling type you just have to change the comic's format from "manga" to "long strip" in the comic's settings.

The front doesn't have all the lyrics as I understand it, 
In the backend and SEO settings would also be nice.

In the admin area itself, you can't specify "long strip" silence.

**Comment by FedericoHeichou** [comment]
Frontend should have everything from what I remember.  
For backend you are right, actually there is not localization.  
I'll add it if there will be enough interest.  

Regarding SEO it is automatic. Do you mean something like customizing meta title and meta description for comic and chapter? You can already manipulate the meta image though (thumbnail link).  

For the "long strip" you can't set it as default in the settings, but you have to specify the format in each comic

**Comment by thefoxcc** [comment]
> Frontend should have everything from what I remember. For backend you are right, actually there is not localization. I'll add it if there will be enough interest.
> 
> Regarding SEO it is automatic. Do you mean something like customizing meta title and meta description for comic and chapter? You can already manipulate the meta image though (thumbnail link).
> 
> For the "long strip" you can't set it as default in the settings, but you have to specify the format in each comic



All right. Uh-huh,

by SEO, I mean manipulation:
https://www.phoenixscans.com/comics/addio-eri
![image](https://github.com/FedericoHeichou/PizzaReader/assets/94108436/3a5d17d8-838c-4387-a946-75107693899f)
https://www.phoenixscans.com/read/oshi-no-ko/it/vol/15/ch/145#1
![image](https://github.com/FedericoHeichou/PizzaReader/assets/94108436/0fe51dc9-b6f0-4910-b12d-ed268139c621)

Oh and another idea is to make a schema and display the manga ranking in the search engine

**Comment by miish3lov** [comment]
Since this issue is still open I'll comment here.

When attempting to translate the UI to Arabic, the characters show as '??????????', perhaps something to do with the database character set? The chapter titles can display arabic characters just fine. The issue appears with menu tabs and 'Last Releases' And all the UI text that is not entered by the user. 

Also, even after changing the comic format to **Long Strip (Web Toons)** the Page Rendering Mode is still defaults to **Single Page** for new readers.

It would be nice to have a way to default it to Long Strip for new users as it is the preferable option for many of my users.

**Comment by FedericoHeichou** [comment]
> Since this issue is still open I'll comment here.
> 
> When attempting to translate the UI to Arabic, the characters show as '??????????', perhaps something to do with the database character set? The chapter titles can display arabic characters just fine. The issue appears with menu tabs and 'Last Releases' And all the UI text that is not entered by the user.
> 
> Also, even after changing the comic format to **Long Strip (Web Toons)** the Page Rendering Mode is still defaults to **Single Page** for new readers.
> 
> It would be nice to have a way to default it to Long Strip for new users as it is the preferable option for many of my users.

Hi
For the long strip seems a bug. Can you open a  separate issue for that?  
For the "????" the text should be set in .js, can you send me an example file?

**Comment by miish3lov** [comment]
Wow, the latest update solved both problems for me, thanks :)

**Comment by FedericoHeichou** [comment]
> Wow, the latest update solved both problems for me, thanks :)
> 
> I guess my final 'localization' request would be to add support for RTL layout.

What do you mean? The RTL is already there, but is there any problem with translations?

**Comment by miish3lov** [comment]
Oh 

> > Wow, the latest update solved both problems for me, thanks :)
> > I guess my final 'localization' request would be to add support for RTL layout.
> 
> What do you mean? The RTL is already there, but is there any problem with translations?

Oh the translation is perfectly fine. 

I meant (header for example)  being in this order from right to left: (Logo - Website name - Menus - Dark/Light mode - Search bar).

I took a screenshot to elaborate further what I mean, I hope it makes it more clear

![Image](https://github.com/user-attachments/assets/69d7b9a1-9499-4db1-bd3c-a3447891c97c)

**Comment by FedericoHeichou** [comment]
> Oh
> 
> > > Wow, the latest update solved both problems for me, thanks :)
> > > I guess my final 'localization' request would be to add support for RTL layout.
> > 
> > 
> > What do you mean? The RTL is already there, but is there any problem with translations?
> 
> Oh the translation is perfectly fine.
> 
> I meant (header for example) being in this order from right to left: (Logo - Website name - Menus - Dark/Light mode - Search bar).
> 
> Another example for the manga info page, I took a screenshot to elaborate further what I mean, I hope it makes it more clear
> 
> ![Image](https://github.com/user-attachments/assets/69d7b9a1-9499-4db1-bd3c-a3447891c97c)
> 
> I already examined .env and didn't find anything related to changing the layout to rtl

Oh, I understand. Didn't know it was a thing!  
Is this an Arabic thing? Like having RTL layout to follow the writing?  
Anyway I will try to have a look, I think it can be done with some CSS trick, but I don't remember how many flex div I have. Surely the bottom table is easy, for other things I have to check.  
Just to not forget it, can you please open a issue for this feature request?

**Comment by miish3lov** [comment]
Yeah. Not just an Arabic thing, more like the natural layout that flow better with any RTL language. Thanks for paying attention to this.
